tests/config/baseTest.ts

```diff
@@ -57,33 +62,48 @@ class DriverMode {
 }
 
 class ServiceMode {
-  private _playwrightObejct: any;
+  private _playwrightObject: any;
   private _client: any;
   private _serviceProcess: childProcess.ChildProcess;
+  private _tmpConfigFile?: string;
 
-  async setup(workerIndex: number) {
-    const port = 10507 + workerIndex;
-    this._serviceProcess = childProcess.fork(path.join(__dirname, '..', '..', 'lib', 'cli', 'cli.js'), ['run-server', String(port)], {
-      stdio: 'pipe'
+  async setup(workerIndex: number, options?: PlaywrightModeSetupOptions) {
+    if (options?.acceptForwardedPorts) {
+      this._tmpConfigFile = path.join(os.tmpdir(), `pw-server-config-${workerIndex}.json`);
+      const config = {acceptForwardedPorts: options.acceptForwardedPorts};
+      await fs.promises.writeFile(this._tmpConfigFile, JSON.stringify(config));
```

Comment:
Why not pass a specific port as before?

Reply:
I wanted to have them auto-generated via listen(0) on the server side, thought it might be better.

Reply:
I still think that hardcoded port was better - gives less room for errors.